packages/instrumentation-mongoose/src/mongoose.ts

```diff
@@ -0,0 +1,268 @@
+    static readonly component = 'mongoose';
+    protected _config: MongooseInstrumentationConfig;
+
+    constructor(config: Config = {}) {
+        super('opentelemetry-instrumentation-mongoose', VERSION, Object.assign({}, config));
+
+        // According to specification, statement is not set by default on mongodb spans.
+        if (!config.dbStatementSerializer) this._config.dbStatementSerializer = () => undefined;
+    }
+
+    setConfig(config: Config = {}) {
+        this._config = Object.assign({}, config);
+        if (!config.dbStatementSerializer) this._config.dbStatementSerializer = () => undefined;
+        if (config.logger) this._logger = config.logger;
+    }
+
+    protected init(): InstrumentationModuleDefinition<typeof mongoose> {
+        const module = new InstrumentationNodeModuleDefinition<typeof mongoose>(
+            MongooseInstrumentation.component,
+            ['*'],
+            this.patch.bind(this),
+            this.unpatch.bind(this)
+        );
+        return module;
+    }
+
+    protected patch(moduleExports: typeof mongoose) {
+        this._logger.debug('mongoose instrumentation: patching');
+
+        this._wrap(moduleExports.Model.prototype, 'save', this.patchOnModelMethods('save'));
+        this._wrap(moduleExports.Model.prototype, 'remove', this.patchOnModelMethods('remove'));
+        this._wrap(moduleExports.Query.prototype, 'exec', this.patchQueryExec());
+        this._wrap(moduleExports.Aggregate.prototype, 'exec', this.patchAggregateExec());
+
+        contextCaptureFunctions.forEach((funcName: string) => {
+            this._wrap(moduleExports.Query.prototype, funcName as any, this.patchAndCaptureSpanContext(funcName));
+        });
+        this._wrap(moduleExports.Model, 'aggregate', this.patchModelAggregate());
+
+        return moduleExports;
+    }
+
+    protected unpatch(moduleExports: typeof mongoose): void {
+        this._logger.debug('mongoose instrumentation: unpatch mongoose');
+        this._unwrap(moduleExports.Model.prototype, 'save');
+        this._unwrap(moduleExports.Model.prototype, 'remove');
+        this._unwrap(moduleExports.Query.prototype, 'exec');
+        this._unwrap(moduleExports.Aggregate.prototype, 'exec');
+
+        contextCaptureFunctions.forEach((funcName: string) => {
+            this._unwrap(moduleExports.Query.prototype, funcName as any);
+        });
+        this._unwrap(moduleExports.Model, 'aggregate');
+    }
+
+    private patchAggregateExec() {
+        const thisInstrumentation = this;
+        thisInstrumentation._logger.debug('mongoose instrumentation: patched mongoose Aggregate exec prototype');
+        return (originalExec: Function) => {
+            return function exec(this: any) {
+                const parentSpan = this[_STORED_PARENT_SPAN];
+                const attributes = {
+                    [DatabaseAttribute.DB_STATEMENT]: thisInstrumentation._config.dbStatementSerializer('aggregate', {
```

Comment:
what happens if `dbStatementSerializer` returns undefined?
Will we clean this attribute from the span at some point?

Reply:
according to the unit tests it's cleared automartically